src/Discord.Net.Commands/Builders/ParameterBuilder.cs

```diff
@@ -56,11 +56,27 @@ internal void SetType(Type type)
 
         private TypeReader GetReader(Type type)
         {
-            var readers = Command.Module.Service.GetTypeReaders(type);
+            var commands = Command.Module.Service;
+            if (type.GetTypeInfo().GetCustomAttribute<NamedArgumentTypeAttribute>() != null)
+            {
+                IsRemainder = true;
+                var reader = commands.GetTypeReaders(type)?.FirstOrDefault().Value;
+                if (reader == null)
+                {
+                    var readerType = typeof(NamedArgumentTypeReader<>).MakeGenericType(new[] { type });
```

Comment:
This would throw if the provided type argument doesn't fit in the constraints. Do we want that exception to bubble up to user-code? I've leaned towards 'yes', but I dunno if we want to change it with a better message or something.